### ISSUE TITLE
Accepting numpy arrays of objects

### ISSUE THREAD

**Issue by jwass** [Toblerity/Shapely#226]
Shapely functions that accept lists of geometries should also be able to accept numpy arrays of geometries.

For example: `LineString(np.array([Point(0.0, 0.0), Point(1.0, 1.0)], dtype='object'))`. Currently this fails as the code that uses `__array_interface__` expects the values to be numeric. We'd just need to check for object dtype and then fall back to the list-like iteration that's already there.

This will better integrate Shapely with GeoPandas as `GeoSeries` is backed by an object array. So if you have a `GeoDataFrame` with `Point` geometry objects, you would be able to just do `LineString(df.geometry)`. The `shapely.ops` collection operators would support this also. If I have a `GeoDataFrame` of roads that have a `name` property, I could run something like: `roads.groupby('name')['geometry'].apply(linemerge)` - which would run `linemerge` on each group of roads that share the same name. Pretty powerful for not much code. We could do this today but you'd have to convert to a `list` everywhere.

Here's what I think has to change:
- `geom_*_from_py()` methods
- `shapely.ops.CollectionOperator` methods
- speedups - as appropriate

Any objections or any issues I missed before I take this on?

**Comment by sgillies** [comment]
None, @jwass. :+1:

**Comment by springcoil** [comment]
This is really cool - was there any progress. I wish I had the time to contribute more.

**Comment by jwass** [comment]
Hi @springcoil. I actually got most of this going when I first posted but found it to be a bit slower than the existing implementation, and then never got back to it. It's almost certainly out of date now. 

I think a better approach would be providing a classmethod like `from_objects()` or something better named - that would be a bit slower, due to type/instance checks and others, but be much more flexible in the input. That way we could stay out of the way when speed is critical, but still maintain the flexibility here.
